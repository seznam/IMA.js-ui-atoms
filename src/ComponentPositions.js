import { UserAgent } from '@ima/plugin-useragent';

/**
 * A bounding client rectangle.
 * @typedef {{
 *            top: number,
 *            left: number,
 *            width: number,
 *            height: number
 *          }} BoundingClientRect
 */

/**
 * Component positions helper.
 */
export default class ComponentPositions {
  static get $dependencies() {
    return ['$Window', UserAgent];
  }

  /**
   * Initializes the helper.
   *
   * @param {ima.window.Window} window
   * @param {UserAgent} userAgent
   */
  constructor(window, userAgent) {
    /**
     * @type {ima.window.Window}
     */
    this._window = window;

    /**
     * @type {UserAgent}
     */
    this._userAgent = userAgent;
  }

  /**
   * Convert string to number.
   *
   * @param {string} string
   * @return {number}
   */
  convertToNumber(string) {
    let number = 0;

    try {
      number = parseInt(string, 10);
    } catch (e) {
      number = 0;
    }

    return number;
  }

  /**
   * Returns percent of visibility defined area in window viewport.
   *
   * @param {BoundingClientRect} elmRect
   * @return {number} The percent of visibility.
   */
  getPercentOfVisibility(elmRect) {
    if (!elmRect) {
      throw new Error(
        `Element rect is required. Call getBoundingClientRect()` +
          ` method on element or give object with properties { top: number,` +
          ` left: number, width: number, height: number }.`
      );
    }

    if (!this._window.isClient()) {
      return 0;
    }

    let windowViewportRect = this.getWindowViewportRect();

    let intersectionRect = this.getRectsIntersection(
      windowViewportRect,
      elmRect
    );
    let percent =
      ((intersectionRect.width * intersectionRect.height) /
        (elmRect.width * elmRect.height)) *
      100;

    return isNaN(percent) ? 0 : percent;
  }

  /**
   * Returns an intersection rectangle of two defined reactangles.
   *
   * @param {BoundingClientRect} rect1
   * @param {BoundingClientRect} rect2
   * @return {BoundingClientRect} The intersection rectangle.
   */
  getRectsIntersection(rect1, rect2) {
    let top = this.getNumberFromRange(rect2.top, rect1.top, rect1.height);
    let left = this.getNumberFromRange(rect2.left, rect1.left, rect1.width);
    let bottom = this.getNumberFromRange(
      rect2.top + rect2.height,
      rect1.top,
      rect1.height
    );
    let right = this.getNumberFromRange(
      rect2.left + rect2.width,
      rect1.left,
      rect1.width
    );
    let width = right - left;
    let height = bottom - top;

    return { top, left, width, height };
  }

  /**
   * Returns number from defined range, if number is not in defined range return
   * min or max depends on number.
   *
   * @param {number} number
   * @param {number} min
   * @param {number} max
   * @return {number}
   */
  getNumberFromRange(number, min, max) {
    return Math.min(Math.max(number, min), max);
  }

  /**
   * Returns window viewport rect.
   *
   * @return {BoundingClientRect}
   */
  getWindowViewportRect() {
    let win = this._window.getWindow();
    let top = 0;
    let left = 0;
    let width = 0;
    let height = 0;

    if (this._window.isClient()) {
      width = win.innerWidth;
      height = win.innerHeight;
    }

    return { top, left, width, height };
  }

  /**
   * Returns window scroll position.
   *
   * @return {{top: number, left: number}}
   */
  getWindowScrollPosition() {
    let left = window.pageXOffset ? window.pageXOffset : 0;
    let top = window.pageYOffset ? window.pageYOffset : 0;

    return { top, left };
  }

  /**
   * Returns the size of an element and its position relative to the viewport
   * and add extended value to returned rect.
   *
   * @param {Element} element
   * @param {{width: number, height: number}} size
   * @param {number} extended
   * @return {BoundingClientRect}
   */
  getBoundingClientRect(
    element,
    { width, height } = { width: 0, height: 0 },
    extended = 0
  ) {
    if (!element || typeof element.getBoundingClientRect !== 'function') {
      throw new Error(
        `Element rect is required with callable getBoundingClientRect()` +
          ` method on element.`
      );
    }

    const clientRect = element.getBoundingClientRect();
    const isInvisibleElement =
      clientRect.top === 0 &&
      clientRect.left === 0 &&
      clientRect.width === 0 &&
      clientRect.height === 0;

    if (isInvisibleElement) {
      const { top, left, width, height } = clientRect;

      return { top, left, width, height };
    }

    let elmRectStyle = {
      top: clientRect.top - extended,
      left: clientRect.left - extended,
      width: clientRect.width + 2 * extended,
      height: (clientRect.height || height || 0 / width || 0) + 2 * extended,
    };

    return this._fixBoundingClientRectOnIOS(elmRectStyle);
  }

  /**
   * Applies a fix for iOS 8+ bug, where overscroll messes up
   * getBoundingClientRect()'s top value on all iOS webkit based devices:
   * https://github.com/lionheart/openradar-mirror/issues/6233
   *
   * @param {BoundingClientRect} boundingClientRect A bounding client rectangle.
   * @return {BoundingClientRect} A fixed bounding client rectangle.
   */
  _fixBoundingClientRectOnIOS(boundingClientRect) {
    if (this._userAgent.getOSFamily() !== 'iOS') {
      return boundingClientRect;
    }

    const window = this._window.getWindow();
    const maxScrollHeight = this._getMaxScrollHeight();
    const rect = Object.assign({}, boundingClientRect);

    if (window.scrollY < 0 && rect.top > 0) {
      rect.top += window.scrollY;
    } else if (window.scrollY > maxScrollHeight && rect.top < 0) {
      rect.top += window.scrollY - maxScrollHeight;
    }

    return rect;
  }

  /**
   * Returns maximum available scroll height (minus viewport height).
   *
   * @returns {number}
   */
  _getMaxScrollHeight() {
    const window = this._window.getWindow();
    const document = this._window.getDocument();

    return (
      Math.max(
        document.body.scrollHeight || 0,
        document.body.offsetHeight || 0,
        document.documentElement.clientHeight || 0,
        document.documentElement.offsetHeight || 0,
        document.documentElement.scrollHeight || 0
      ) - window.innerHeight
    );
  }
}
