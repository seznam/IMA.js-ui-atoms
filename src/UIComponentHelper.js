import ComponentPositions from './ComponentPositions';
import Visibility from './Visibility';

/**
 * UI component helper.
 */
export default class UIComponentHelper {

	static get $dependencies() {
		return [
			'$Router',
			ComponentPositions,
			Visibility,
			'$CssClasses'
		];
	}

	/**
	 * Initializes the helper.
	 *
	 * @param {ima.router.Router} router
	 * @param {ComponentPositions} componentPositions
	 * @param {Visibility} visibility
	 * @param {function(...?(boolean|string|React.Component|Object<string, boolean>)): string} cssClassNameProcessor
	 */
	constructor(router, componentPositions, visibility, cssClassNameProcessor) {
		/**
		 * IMA Router
		 *
		 * @type {ima.router.Router}
		 */
		this._router = router;

		/**
		 * Component position
		 *
		 * @type {ComponentPosition}
		 */
		this._componentPositions = componentPositions;

		/**
		 * Visibility helper
		 *
		 * @type {Visibility}
		 */
		this._visibility = visibility;

		/**
		 * @type {function(...?(boolean|string|React.Component|Object<string, boolean>)): string}
		 */
		this._cssClassNameProcessor = cssClassNameProcessor;
	}

	/**
	 * The public getter for visibility helper.
	 *
	 * @return {Visibility}
	 */
	get visibility() {
		return this._visibility;
	}

	/**
	 * The public getter for component positions helper.
	 *
	 * @return {ComponentPositions}
	 */
	get componentPositions() {
		return this._componentPositions;
	}

	/**
	 * Returns true if page may be rendered as amp page.
	 *
	 * @return {boolean}
	 */
	isAmp() {
		let ampParam = null;

		try {
			ampParam = this._router.getCurrentRouteInfo().params.amp;
		} catch (error) {
			ampParam = false;
		}

		return ampParam === true || ampParam === '1';
	}

	/**
	 * Filters the provided properties and returns only the properties which's
	 * names start with the {@code data-} prefix.
	 *
	 * @param {Object<string, *>} props
	 * @return {Object<string, (number|string)>}
	 */
	getDataProps(props) {
		let dataProps = {};

		for (let propertyName of Object.keys(props)) {
			if (/^data-/.test(propertyName)) {
				dataProps[propertyName] = props[propertyName];
			}
		}

		return dataProps;
	}

	/**
	 * Filters the provided properties and returns only the properties which's
	 * names start with the {@code aria-} prefix.
	 *
	 * @param {Object<string, *>} props
	 * @return {Object<string, (number|string)>}
	 */
	getAriaProps(props) {
		let ariaProps = {};

		for (let propertyName of Object.keys(props)) {
			if (/^aria-/.test(propertyName)) {
				ariaProps[propertyName] = props[propertyName];
			}
		}

		return ariaProps;
	}

	/**
	 * Generate a string of CSS classes from the properties of the passed-in
	 * object that resolve to true.
	 *
	 * @param {...?(string|Object<string, boolean>)} classRuleGroups CSS
	 *        classes in a string separated by whitespace, or a map of CSS
	 *        class names to boolean values. The CSS class name will be
	 *        included in the result only if the value is {@code true}.
	 *        Declarations in the later class rule group will override the
	 *        declarations in the previous group.
	 * @return {string} String of CSS classes that had their property resolved
	 *         to {@code true}.
	 */
	cssClasses(...classRuleGroups) {
		return this._cssClassNameProcessor(...classRuleGroups);
	}

	/**
	 * Only Facade mehtod of ComponentPositions.convertToNumber function
	 * for backward compatibility.
	 *
	 * @deprecated removed with next version
	 * @param {string} string
	 * @return {number}
	 */
	convertToNumber(string) {
		return this._componentPositions.convertToNumber(string);
	}

	/**
	 * Only Facade mehtod of ComponentPositions.getWindowViewportRect function
	 * for backward compatibility.
	 *
	 * @deprecated removed with next version
	 * @return {{top: number, left: number, width: number, height: number}}
	 */
	getWindowViewportRect() {
		return this._componentPositions.getWindowViewportRect();
	}

	/**
	 * Only Facade mehtod of ComponentPositions.getBoundingClientRect function
	 * for backward compatibility.
	 *
	 * @deprecated removed with next version
	 * @param {Element} element
	 * @param {{width: number, height: number}} size
	 * @param {number} extended
	 * @return {{top: number, left: number, width: number, height: number}}
	 */
	getBoundingClientRect(element, size, extended) {
		return this._componentPositions.getBoundingClientRect(element, size, extended);
	}

	/**
	 * Only Facade mehtod of ComponentPositions.getPercentOfVisibility function
	 * for backward compatibility.
	 *
	 * @deprecated removed with next version
	 * @param {{top: number, left: number, width: number, height: number}} elmRect
	 * @return {number} The percent of visibility.
	 */
	getPercentOfVisibility(elmRect) {
		return this._componentPositions.getPercentOfVisibility(elmRect);
	}

	/**
	 * @param {HTMLElement} element
	 * @param {{ width: ?number, height: ?number, extendedPadding: ?number }} options
	 * @return {function}
	 */
	getVisibilityReader(element, options) {
		let self = this;

		return function readVisibility() {
			let elementRect = self._componentPositions.getBoundingClientRect(
				element,
				{ width: options.width, height: options.height },
				options.extendedPadding
			);

			return self._componentPositions.getPercentOfVisibility(elementRect);
		};
	}

	/**
	 * Only Facade mehtod of Visibility.throttle function
	 * for backward compatibility.
	 *
	 * @deprecated removed with next version
	 * @param {function(...)} eventHandler
	 * @param {number?} interval
	 * @param {Object?} context
	 * @return {function(...)} The throttled event
	 */
	throttle(eventHandler, interval, context) {
		return this._visibility.throttle(eventHandler, interval, context);
	}

	/**
	 * Register component to main visibility loop
	 *
	 * @param {function} reader
	 * @param {function} writer
	 * @param {{ visibilityInterval: number }} options
	 * @return {number} The registered id
	 */
	registerComponentToVisibility(reader, writer, options = { visibilityInterval: 180 }) {
		return this._visibility.register(reader, writer, options);
	}

	/**
	 * Unregister component from main visibility loop
	 *
	 * @param {number} visibilityId
	 */
	unregisterComponentToVisibility(visibilityId) {
		this._visibility.unregister(visibilityId);
	}
}
