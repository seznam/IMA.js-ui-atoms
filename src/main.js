import { Infinite, Circle, uuid } from 'infinite-circle';

import UIComponentHelper from './UIComponentHelper';
import ComponentPositions from './ComponentPositions';
import Visibility from './Visibility';

import Headline from './headline/Headline';
import Headline1 from './headline/Headline1';
import Headline2 from './headline/Headline2';
import Headline3 from './headline/Headline3';
import Headline4 from './headline/Headline4';
import Headline5 from './headline/Headline5';
import Headline6 from './headline/Headline6';

import Iframe from './iframe/Iframe';

import Image from './image/Image';

import Link from './link/Link';

import List from './list/List';
import OrderedList from './list/OrderedList';
import UnorderedList from './list/UnorderedList';
import ListItem from './list/ListItem';

import Loader from './loader/Loader';

import Paragraph from './paragraph/Paragraph';

import Sizer from './sizer/Sizer';

import Video from './video/Video';

const defaultDependencies = ['$Router', ComponentPositions, Visibility];

const $registerImaPlugin = ns => {
  ns.namespace('ima.ui.atom');
  ns.namespace('ima.ui.atom.headline');
  ns.namespace('ima.ui.atom.iframe');
  ns.namespace('ima.ui.atom.image');
  ns.namespace('ima.ui.atom.link');
  ns.namespace('ima.ui.atom.list');
  ns.namespace('ima.ui.atom.loader');
  ns.namespace('ima.ui.atom.paragraph');
  ns.namespace('ima.ui.atom.sizer');
  ns.namespace('ima.ui.atom.video');

  const nsHeadline = ns.ima.ui.atom.headline;
  nsHeadline.Headline = Headline;
  nsHeadline.Headline1 = Headline1;
  nsHeadline.Headline2 = Headline2;
  nsHeadline.Headline3 = Headline3;
  nsHeadline.Headline4 = Headline4;
  nsHeadline.Headline5 = Headline5;
  nsHeadline.Headline6 = Headline6;
  nsHeadline.H1 = Headline1;
  nsHeadline.H2 = Headline2;
  nsHeadline.H3 = Headline3;
  nsHeadline.H4 = Headline4;
  nsHeadline.H5 = Headline5;
  nsHeadline.H6 = Headline6;

  ns.ima.ui.atom.iframe.Iframe = Iframe;

  ns.ima.ui.atom.image.Image = Image;
  ns.ima.ui.atom.image.Img = Image;

  ns.ima.ui.atom.link.Link = Link;
  ns.ima.ui.atom.link.A = Link;

  const nsList = ns.ima.ui.atom.list;
  nsList.List = List;
  nsList.ListItem = ListItem;
  nsList.OrderedList = OrderedList;
  nsList.UnorderedList = UnorderedList;
  nsList.Li = ListItem;
  nsList.Ol = OrderedList;
  nsList.Ul = UnorderedList;

  ns.ima.ui.atom.loader.Loader = Loader;

  ns.ima.ui.atom.paragraph.Paragraph = Paragraph;
  ns.ima.ui.atom.paragraph.P = Paragraph;

  ns.ima.ui.atom.sizer.Sizer = Sizer;

  ns.ima.ui.atom.video.Video = Video;

  ns.ima.ui.atom.UIComponentHelper = UIComponentHelper;
  ns.ima.ui.atom.Visibility = Visibility;
  ns.ima.ui.atom.ComponentPositions = ComponentPositions;
  ns.ima.ui.atom.defaultDependencies = defaultDependencies;
};

const initBind = (ns, oc) => {
  oc.inject(Infinite, []);
  //oc.inject(UIComponentHelper, defaultDependencies);
};

const initServices = (ns, oc) => {
  oc.get(UIComponentHelper).init();
};

const initSettings = () => {
  return {
    prod: {
      plugin: {
        imaUiAtoms: {
          useIntersectionObserver: {
            iframes: true,
            images: true,
            videos: true
          },
          disableNoScript: {
            iframes: false,
            images: false,
            videos: false
          }
        }
      }
    }
  };
};

export {
  UIComponentHelper,
  Visibility,
  ComponentPositions,
  defaultDependencies,
  Headline,
  Headline1,
  Headline2,
  Headline3,
  Headline4,
  Headline5,
  Headline6,
  Headline1 as H1,
  Headline2 as H2,
  Headline3 as H3,
  Headline4 as H4,
  Headline5 as H5,
  Headline6 as H6,
  Iframe,
  Image,
  Image as Img,
  Link,
  Link as A,
  List,
  ListItem,
  OrderedList,
  UnorderedList,
  ListItem as Li,
  OrderedList as Ol,
  UnorderedList as Ul,
  Loader,
  Paragraph,
  Paragraph as P,
  Sizer,
  Video,
  initBind,
  initServices,
  initSettings,
  $registerImaPlugin,
  Circle,
  Infinite,
  uuid
};
