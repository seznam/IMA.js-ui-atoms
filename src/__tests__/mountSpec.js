import { mount } from 'enzyme';
import React from 'react';
import PropTypes from 'prop-types';
import { toMockedInstance } from 'to-mock';
import classnames from 'classnames';

import * as UIAtoms from '../main';
import UIComponentHelper from '../UIComponentHelper';
import ComponentPositions from '../ComponentPositions';
import Visibility from '../Visibility';
import { Infinite } from 'infinite-circle';

import { JSDOM } from 'jsdom';
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;

import _router from '../mocks/router';
import _window from '../mocks/window';
import _settings from '../mocks/settings';

global.window = window;
global.document = window.document;
global.navigator = {
  userAgent: 'node.js'
};

const visibility = toMockedInstance(Visibility);
const mockPosition = {
  height: 0
};
const componentPositions = toMockedInstance(ComponentPositions, {
  getWindowViewportRect: () => mockPosition
});
const infinite = toMockedInstance(Infinite);

const childContextTypes = {
  $Utils: PropTypes.shape({}),
  $Settings: PropTypes.shape({})
};

function getComponentOptions(overrideSettings = {}) {
  const $Settings = Object.assign({}, _settings, overrideSettings);

  const $UIComponentHelper = new UIComponentHelper(
    _router,
    _window,
    componentPositions,
    visibility,
    infinite,
    classnames,
    $Settings
  );

  const context = {
    $Utils: {
      $Settings,
      $UIComponentHelper
    }
  };
  const mountOptions = {
    context,
    childContextTypes,
    someProp: true
  };

  return mountOptions;
}

describe('UIAtoms mount rendering', () => {
  let wrapper = null;

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
    }
  });

  describe('should render atoms with noscript tag: ', () => {
    let mountOptions = getComponentOptions();

    it('should render Image with noscript tag', () => {
      const Component = UIAtoms.Image;
      wrapper = mount(<Component src="example.jpg" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });

    it('should render Video with noscript tag', () => {
      const Component = UIAtoms.Video;
      wrapper = mount(<Component src="example.mov" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });

    it('should render Iframe with noscript tag', () => {
      const Component = UIAtoms.Iframe;
      wrapper = mount(<Component src="example.html" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });
  });

  describe('should render atoms without noscript tag - disabled by props: ', () => {
    let mountOptions = getComponentOptions();

    it('Image without noscript tag', () => {
      const Component = UIAtoms.Image;
      wrapper = mount(
        <Component src="example.jpg" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });

    it('Video without noscript tag', () => {
      const Component = UIAtoms.Video;
      wrapper = mount(
        <Component src="example.mov" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });

    it('Iframe without noscript tag', () => {
      const Component = UIAtoms.Iframe;
      wrapper = mount(
        <Component src="example.html" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });
  });

  describe('should render atoms without noscript tag - disabled by $Settings: ', () => {
    let mountOptions = getComponentOptions({
      plugin: {
        imaUiAtoms: {
          useIntersectionObserver: {
            iframes: true,
            images: true,
            videos: true
          },
          disableNoScript: {
            iframes: true,
            images: true,
            videos: true
          }
        }
      }
    });

    it('Image without noscript tag', () => {
      const Component = UIAtoms.Image;
      wrapper = mount(<Component src="example.jpg" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });
    it('Video without noscript tag', () => {
      const Component = UIAtoms.Video;
      wrapper = mount(<Component src="example.mov" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });

    it('Iframe without noscript tag', () => {
      const Component = UIAtoms.Iframe;
      wrapper = mount(<Component src="example.html" />, mountOptions);

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });
  });

  describe('should render atoms with noscript tag - disabled by $Settings, overriden by props: ', () => {
    let mountOptions = getComponentOptions({
      plugin: {
        imaUiAtoms: {
          useIntersectionObserver: {
            iframes: true,
            images: true,
            videos: true
          },
          disableNoScript: {
            iframes: true,
            images: true,
            videos: true
          }
        }
      }
    });

    it('Image without noscript tag', () => {
      const Component = UIAtoms.Image;
      wrapper = mount(
        <Component src="example.jpg" disableNoScript={false} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });

    it('Video without noscript tag', () => {
      const Component = UIAtoms.Video;
      wrapper = mount(
        <Component src="example.mov" disableNoScript={false} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });

    it('Iframe without noscript tag', () => {
      const Component = UIAtoms.Iframe;
      wrapper = mount(
        <Component src="example.html" disableNoScript={false} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(1);
    });
  });

  describe('should render atoms without noscript tag - enabled by $Settings, overriden by props: ', () => {
    let mountOptions = getComponentOptions({
      plugin: {
        imaUiAtoms: {
          useIntersectionObserver: {
            iframes: true,
            images: true,
            videos: true
          },
          disableNoScript: {
            iframes: false,
            images: false,
            videos: false
          }
        }
      }
    });

    it('Image without noscript tag', () => {
      const Component = UIAtoms.Image;
      wrapper = mount(
        <Component src="example.jpg" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });

    it('Video without noscript tag', () => {
      const Component = UIAtoms.Video;
      wrapper = mount(
        <Component src="example.mov" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });

    it('Iframe without noscript tag', () => {
      const Component = UIAtoms.Iframe;
      wrapper = mount(
        <Component src="example.html" disableNoScript={true} />,
        mountOptions
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('noscript').length).toEqual(0);
    });
  });
});
