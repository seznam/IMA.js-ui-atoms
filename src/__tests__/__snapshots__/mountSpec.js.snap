// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIAtoms mount rendering should render atoms with noscript tag - disabled by $Settings, overriden by props:  Iframe without noscript tag 1`] = `
"<div class=\\"atm-iframe atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript class=\\"atm-fill\\" style=\\"display: block; width: auto; height: auto;\\"><iframe
                  src=\\"example.html\\"
                  
                  width=\\"auto\\"
                  height=\\"auto\\"
                  
                  scrolling=\\"no\\"
                  frameborder=\\"0\\"
                                  
                  allowfullscreen=\\"0\\"
                  
                  
                  class=\\"atm-fill atm-loaded\\"
                  ></iframe></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms with noscript tag - disabled by $Settings, overriden by props:  Image without noscript tag 1`] = `
"<div class=\\"atm-image atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript><img
                  src=\\"example.jpg\\"
                  srcset=\\"\\"
                  sizes=\\"\\"
                  alt=\\"\\"
                  class=\\"atm-fill atm-loaded\\"
                  /></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms with noscript tag - disabled by $Settings, overriden by props:  Video without noscript tag 1`] = `
"<div class=\\"atm-video atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript><video
								src=\\"example.mov\\"
								poster=\\"\\"
								controls
								
								
								
								class=\\"atm-fill atm-loaded\\"
								></video></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms with noscript tag:  should render Iframe with noscript tag 1`] = `
"<div class=\\"atm-iframe atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript class=\\"atm-fill\\" style=\\"display: block; width: auto; height: auto;\\"><iframe
                  src=\\"example.html\\"
                  
                  width=\\"auto\\"
                  height=\\"auto\\"
                  
                  scrolling=\\"no\\"
                  frameborder=\\"0\\"
                                  
                  allowfullscreen=\\"0\\"
                  
                  
                  class=\\"atm-fill atm-loaded\\"
                  ></iframe></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms with noscript tag:  should render Image with noscript tag 1`] = `
"<div class=\\"atm-image atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript><img
                  src=\\"example.jpg\\"
                  srcset=\\"\\"
                  sizes=\\"\\"
                  alt=\\"\\"
                  class=\\"atm-fill atm-loaded\\"
                  /></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms with noscript tag:  should render Video with noscript tag 1`] = `
"<div class=\\"atm-video atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"><noscript><video
								src=\\"example.mov\\"
								poster=\\"\\"
								controls
								
								
								
								class=\\"atm-fill atm-loaded\\"
								></video></noscript></div>"
`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by $Settings:  Iframe without noscript tag 1`] = `"<div class=\\"atm-iframe atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by $Settings:  Image without noscript tag 1`] = `"<div class=\\"atm-image atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by $Settings:  Video without noscript tag 1`] = `"<div class=\\"atm-video atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by props:  Iframe without noscript tag 1`] = `"<div class=\\"atm-iframe atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by props:  Image without noscript tag 1`] = `"<div class=\\"atm-image atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - disabled by props:  Video without noscript tag 1`] = `"<div class=\\"atm-video atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - enabled by $Settings, overriden by props:  Iframe without noscript tag 1`] = `"<div class=\\"atm-iframe atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - enabled by $Settings, overriden by props:  Image without noscript tag 1`] = `"<div class=\\"atm-image atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;

exports[`UIAtoms mount rendering should render atoms without noscript tag - enabled by $Settings, overriden by props:  Video without noscript tag 1`] = `"<div class=\\"atm-video atm-overflow atm-placeholder\\" style=\\"width: auto; height: auto;\\"></div>"`;
